deno-installer.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+set -u
+
+main() {
+
+  need_cmd downloader --check
+  need_cmd uname
+  need_cmd mktemp
+  need_cmd chmod
+  need_cmd mkdir
+  need_cmd rm
+  need_cmd rmdir
+  need_cmd jq
+
+  get_architecture || return 1
+
+  local _release_name=deno_"$RETVAL"_x64
+
+  get_release_url "$_release_name"
+
+  local _ext=".gz"
+  case "$RETVAL" in
+  *win*)
+      _ext=".zip"
+      ;;
+  esac
+
+  local _dir="$(mktemp -d 2>/dev/null || ensure mktemp -d -t deno_installer)"
+  local _file="${_dir}/deno_${RETVAL}_x64${_ext}"
+
+  ensure downloader $RELEASE_URL $_file
+
+  local _deno_home="${HOME}/deno"
+  local _deno_bin="$_deno_home/deno"
```

Comment:
Maybe install into `$HOME/.deno/bin` ?

Reply:
The user will need to be told to add the PATH to their .bash_profile after install. 
You don't need to do that automatically.

Reply:
maybe make this public as `DENO_HOME` (similar to `RUSTUP_HOME`), and default to XDG_CONFIG_HOME, and fall back to $HOME/.deno ?

Reply:
Ok. Will do!

Reply:
@ry @jedahan This is changed so that DENO_HOME="$HOME/.deno/"
I can try the XDG_CONFIG_HOME and then fallback if that's something we need to have.

Reply:
Just use `$HOME/.deno`... I don't know about `$XDG_CONFIG_HOME` sounds like some unnecessary complexity.